x/evidence/keeper/infraction.go

```diff
@@ -23,10 +23,31 @@ import (
 //
 // TODO: Some of the invalid constraints listed above may need to be reconsidered
 // in the case of a lunatic attack.
+
 func (k Keeper) handleEquivocationEvidence(ctx sdk.Context, evidence *types.Equivocation) {
 	logger := k.Logger(ctx)
 	consAddr := evidence.GetConsensusAddress()
 
+	if _, err := k.slashingKeeper.GetPubkey(ctx, consAddr.Bytes()); err != nil {
+		// Ignore evidence that cannot be handled.
+		//
+		// NOTE: We used to panic with:
+		// `panic(fmt.Sprintf("Validator consensus-address %v not found", consAddr))`,
+		// but this couples the expectations of the app to both CometBFT and
+		// the simulator.  Both are expected to provide the full range of
+		// allowable but none of the disallowed evidence types.  Instead of
+		// getting this coordination right, it is easier to relax the
+		// constraints and ignore evidence that cannot be handled.
+		newConsAddr, _, err := k.slashingKeeper.GetMappedPubkey(ctx, consAddr.Bytes())
```

Comment:
Thanks @tac0turtle. Can this code be moved bellow inside `if !validator.GetOperator().Empty() { } `?

Reply:
> Can this code be moved bellow inside if !validator.GetOperator().Empty() { } ?
I'm not sure if we can move this down to `if !validator.GetOperator().Empty() { }` , before this `if` block we are trying to fetch the validatorByConsAddr, which may return `nil` for validators which already rotated their keys and satisfies the condition to be returned.

Reply:
Oh I see. Then, can we just avoid returning in this if? i.e.,
```golang
if _, err := k.slashingKeeper.GetPubkey(ctx, consAddr.Bytes()); err != nil {
  newConsAddr, _, err := k.slashingKeeper.GetMappedPubkey(ctx, consAddr.Bytes())
  if err == nil && newConsAddr != nil  {
    // take the rotated key
    consAddr = newConsAddr
  } 
}
```
If there is a problem with `GetPubkey` (which is the only thing that can error in GetMappedPubkey), then it will be handle in `if !validator.GetOperator().Empty()`.